notebooks/MAST/HSC/CASJOBS/HSCV3_SMC_API/hscv3_smc_api.ipynb

```diff
@@ -0,0 +1,516 @@
+    "tab['V-I'] = tab['A_F555W'] - tab['A_F814W']\n",
+    "tab = tab[(tab['V-I'] < 1.5) & (tab['V-I'] > -1.5)]\n",
+    "print(f\"{(time.time()-t0):.1f} s: selected {len(tab)} objects with -1.5 < V-I < 1.5\")\n",
+    "\n",
+    "# clean up the output format\n",
+    "tab['A_F555W'].format = \"{:.3f}\"\n",
+    "tab['A_F814W'].format = \"{:.3f}\"\n",
+    "tab['V-I'].format = \"{:.3f}\"\n",
+    "tab['CI'].format = \"{:.3f}\"\n",
+    "tab['MatchRA'].format = \"{:.6f}\"\n",
+    "tab['MatchDec'].format = \"{:.6f}\"\n",
+    "tab"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Plot object positions on the sky <a name=\"positions\"></a>\n",
+    "\n",
+    "We mark the galaxy center as well.  These fields are sprinkled all over the galaxy (as determined by the HST proposals)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(1, (10, 10))\n",
+    "plt.plot(tab['MatchRA'], tab['MatchDec'], 'bo', markersize=1, label=f'{len(tab)} HSC measurements')\n",
+    "plt.plot(ra, dec, 'rx', label=target, markersize=10)\n",
+    "plt.gca().invert_xaxis()\n",
+    "plt.gca().set_aspect(1.0/np.cos(np.radians(dec)))\n",
+    "plt.xlabel('RA [deg]')\n",
+    "plt.ylabel('Dec [deg]')\n",
+    "plt.legend(loc='best')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Plot the color-magnitude diagram <a name=\"cmd\"></a>\n",
+    "\n",
+    "This uses the `fastkde` module to get a kernel density estimate in order to plot a dense scatterplot."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate the point density\n",
+    "t0 = time.time()\n",
+    "x = tab['V-I']\n",
+    "y = tab['A_F555W']\n",
+    "myPDF, axes = fastKDE.pdf(x, y, numPoints=2**9+1)\n",
+    "print(f\"kde took {(time.time()-t0):.1f} sec for {len(x)} points\")\n",
+    "\n",
+    "# interpolate to get z values at points\n",
+    "finterp = RegularGridInterpolator((axes[1], axes[0]), myPDF, bounds_error=False, fill_value=0.0)\n",
+    "z = finterp((y, x))\n",
+    "\n",
+    "# Sort the points by density, so that the densest points are plotted last\n",
+    "idx = z.argsort()\n",
+    "xs, ys, zs = x[idx], y[idx], z[idx]\n",
+    "\n",
+    "plt.figure(1, (12, 10))\n",
+    "\n",
+    "# select a subset of points to plot\n",
+    "wsel = select_subset(zs,axes,verbose=True)\n",
+    "\n",
+    "plt.scatter(xs[wsel], ys[wsel], c=zs[wsel], s=2, edgecolors='none', cmap='plasma')\n",
+    "plt.ylabel('V [mag]')\n",
+    "plt.xlabel('V - I [mag]')\n",
+    "plt.xlim(-1.5, 1.5)\n",
+    "plt.ylim(14, 27)\n",
+    "plt.gca().invert_yaxis()\n",
+    "plt.title(f'{len(tab):,} stars in the Small Magellanic Cloud')\n",
+    "plt.colorbar()\n",
+    "plt.tight_layout()\n",
+    "plt.savefig(f\"{resPath}/smc_cmd.png\")"
```

Comment:
```suggestion
    "plt.tight_layout()\n"
```
I'd say drop the savefig.  Some users get confused by plots appearing for some cells and not others.

Reply:
The plot still appears in the notebook

Reply:
done